### ISSUE TITLE
Evaluation using a temporary checkpoint might be erroneous.

### ISSUE THREAD

**Issue by K-H-Ismail** [Thvnvtos/CSNN-1D-Delays#2]
In the eval model function, the state_dict of the training is saved using a temporary checkpoint. The DCLS layers are then forced to max version and sig parameter to 0 before rounding the P parameters.

 https://github.com/Thvnvtos/CSNN-1D-Delays/blob/6d5aaabf5907eeaf178a2d74b0dbbd8d9396f43e/model.py#L158

then the model is evaluated in inference and metrics are calculated.

The eval model function ends by switching back to train mode and erasing the checkpoint.

Evaluating the model this way could prone to errors especially in parallel and distributed setup.

**Comment by K-H-Ismail** [comment]
Solved in PR https://github.com/Thvnvtos/CSNN-1D-Delays/pull/5
